spec/controllers/state_file/archived_intake/mailing_address_validation_controller_spec.rb

```diff
@@ -0,0 +1,107 @@
+        allow(current_request).to receive(:access_locked?).and_return(true)
+      end
+
+      it "redirects to error page" do
+        get :edit
+
+        expect(response).to redirect_to(state_file_archived_intakes_verification_error_path)
+      end
+    end
+
+    context "when the request is not locked" do
+      before do
+        allow(current_request).to receive(:access_locked?).and_return(false)
+      end
+
+      it "renders the edit template with a new MailingAddressValidationForm" do
+        get :edit
+
+        expect(assigns(:form)).to be_a(StateFile::ArchivedIntakes::MailingAddressValidationForm)
+        expect(response).to render_template(:edit)
+      end
+    end
+
+    it "redirect to root if code verification was not completed" do
+      session[:code_verified] = nil
+      session[:ssn_verified] = true
+      get :edit
+
+      expect(response).to redirect_to(root_path)
+      expect(StateFileArchivedIntakeAccessLog.last.event_type).to eq("unauthorized_mailing_attempt")
+    end
+
+    it "redirect to root if ssn verification was not completed" do
+      session[:code_verified] = true
+      session[:ssn_verified] = nil
+      get :edit
+
+      expect(response).to redirect_to(root_path)
+      expect(StateFileArchivedIntakeAccessLog.last.event_type).to eq("unauthorized_mailing_attempt")
+    end
+  end
+
+  describe "PATCH #update" do
+    context "with a valid chosen address" do
+      it "creates an access log and redirects to the root path" do
+        post :update, params: {
+          state_file_archived_intakes_mailing_address_validation_form: { selected_address: intake.full_address, addresses: controller.address_challenge_set}
+        }
+        expect(assigns(:form)).to be_valid
+
+        access_log = StateFileArchivedIntakeAccessLog.last
+        expect(access_log.state_file_archived_intake_request).to eq(current_request)
+        expect(access_log.event_type).to eq("correct_mailing_address")
+
+        # TODO: https://codeforamerica.atlassian.net/browse/FYST-1520
+        # need to change to download path
+        expect(response).to redirect_to(root_path)
+      end
+    end
+
+    context "with an invalid chosen address" do
+      it "creates an access log and redirects to the root path and locks the request" do
+        post :update, params: {
+          state_file_archived_intakes_mailing_address_validation_form: { selected_address: current_request.fake_address_1, addresses: controller.address_challenge_set}
+        }
+        expect(assigns(:form)).not_to be_valid
+
+        access_log = StateFileArchivedIntakeAccessLog.last
+        expect(access_log.state_file_archived_intake_request).to eq(current_request)
+        expect(access_log.event_type).to eq("incorrect_mailing_address")
+        expect(current_request.access_locked?).to eq(true)
+        expect(response).to redirect_to(state_file_archived_intakes_verification_error_path)
+      end
+    end
+
+    context "without a chosen address" do
```

Comment:
can you explain the case when this would happen where a post to update would occur & we wouldn't want to lock them out?

Reply:
handles the case where the user is hits submit without selecting an addresss